### PR TITLE
Refactored USDC component to use redux store instead of useState

### LINKED ISSUE #34: Move USDC component state variables to redux store
We should move the state variables in this component https://github.com/wormhole-foundation/example-token-bridge-ui/blob/main/src/components/USDC/index.tsx into the redux store to make it less cluttered

### PR DESCRIPTION
**Pull request by kev1n-peters**
Resolves #34

### CODE REVIEW

File: src/components/USDC/index.tsx [hunk 1]
```diff
@@ -454,15 +467,15 @@ function USDC() {
       );
       const maxSwap = await contract.calculateMaxSwapAmount(targetAsset);
       if (cancelled) return;
-      setMaxSwapAmount(maxSwap.toBigInt());
+      dispatch(setMaxSwapAmount(maxSwap.toString()));
```
Comment (kev1n-peters) [inline src/components/USDC/index.tsx:470]: redux can't handle the `bigint` type, because it's non-serializable, so we store as a string instead

**Comment by evan-gray** [comment]
Encountering this issue when using the relayer and moving to the redeem step.
```
Uncaught (in promise) Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="decimals()"
Uncaught (in promise) Error: call revert exception [ See: https://links.ethers.org/v5-errors-CALL_EXCEPTION ] (method="relayerFee(uint16,address)"
```

**Review by evan-gray — CHANGES_REQUESTED** [review]
(see comment)

**Review by evan-gray — APPROVED** [review]
did a few transfers and seems to work. not sure what causes the bad `decimals` call (likely a bad effect) but it seems like it might have been there before.

**Pull request merged**